packages/contractkit/src/identity/offchain-data-wrapper.ts

```diff
@@ -36,34 +37,48 @@ class NoStorageRootProvidedData extends RootError<OffchainErrorTypes.NoStorageRo
   }
 }
 
-export type OffchainErrors = FetchError | InvalidSignature | NoStorageRootProvidedData
+class NoStorageProvider extends RootError<OffchainErrorTypes.NoStorageProvider> {
+  constructor() {
+    super(OffchainErrorTypes.NoStorageProvider)
+  }
+}
+
+export type OffchainErrors =
+  | FetchError
+  | InvalidSignature
+  | NoStorageRootProvidedData
+  | NoStorageProvider
 
 export default class OffchainDataWrapper {
   storageWriter: StorageWriter | undefined
+  signer: string
 
-  constructor(readonly self: string, readonly kit: ContractKit) {}
+  constructor(readonly self: string, readonly kit: ContractKit, signer?: string) {
+    this.signer = signer || self
+  }
 
   async readDataFromAsResult(
     account: string,
+    buildTypedData: (x: Buffer) => Promise<EIP712TypedData>,
```

Comment:
Every call site I've seen seems to use the `buildEIP712TypedData` function. Unless there is a compelling reason this needs to be a callback, I would suggest replacing this arg with the data type (i.e. the own that would be passed to `buildEIP712TypedData`) and calling `buildEIP712TypedData` internally.